DNN Platform/DotNetNuke.Web/Common/LazyServiceProvider.cs

```diff
@@ -5,28 +5,30 @@
 namespace DotNetNuke.Web.Common
 {
     using System;
-    using System.Collections.Generic;
-    using System.Linq;
-    using System.Text;
-    using System.Threading.Tasks;
+    using System.ComponentModel;
 
-    public class LazyServiceProvider : IServiceProvider
+    public class LazyServiceProvider : IServiceProvider, INotifyPropertyChanged
     {
-        private IServiceProvider _serviceProvider;
-
+        private IServiceProvider serviceProvider;
+
+        /// <inheritdoc/>
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        /// <inheritdoc/>
         public object GetService(Type serviceType)
         {
-            if (this._serviceProvider is null)
+            if (this.serviceProvider is null)
             {
                 throw new Exception("Cannot resolve services until the service provider is built.");
             }
 
-            return this._serviceProvider.GetService(serviceType);
+            return this.serviceProvider.GetService(serviceType);
         }
 
         internal void SetProvider(IServiceProvider serviceProvider)
         {
-            this._serviceProvider = serviceProvider;
+            this.serviceProvider = serviceProvider;
+            this.PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(serviceProvider)));
```

Comment:
I added `INotifyPropertyChanged` here to allow the `Globals.cs` to listen for an event when the Service Provider is ready for us. Since the `Globals.cs` is wrapping several Dependency Injection calls as we deprecate methods it is important that it can resolve dependencies once they are available. This is critical when using `Globals.cs` at startup.

I decided on `INotifyPropertyChanged` as it is a .NET Standard type and is used in the desktop/mobile space a lot for notifying objects when something has changed. If there is a better option, I am more than happy to change this